app/activeproject.h

```diff
@@ -116,12 +116,18 @@ class ActiveProject: public QObject
 
     //! Returns true if the project has at least one layer that allows recording
     Q_INVOKABLE bool projectHasRecordingLayers() const;
+
     /**
      * Returns role/permission level of current user for this project
      */
     Q_INVOKABLE QString projectRole() const;
     void setProjectRole( const QString &role );
 
+    /**
+     * Returns true if user role is different than reader
+     */
+    Q_INVOKABLE bool userHasEditableRole() const;
```

Comment:
If you want to use this method instead of the `projectRole`, then I suggest creating a new property `userHasEditRights`. Using method is not ideal as you won't be notified about change - your property bindings in QML won't work